dags/iop/parser.py

```diff
@@ -83,9 +79,12 @@ def __init__(self) -> None:
                 required=True,
             ),
             CustomExtractor(
-                destination="copyright",
-                extraction_function=self._get_copyright,
-                required=True,
+                destination="copyright_year",
+                extraction_function=self._extract_copyright_year,
```

Comment:
where is this method now?

Reply:
https://github.com/cern-sis/workflows/blob/5f3a04512c90045cc1898f3bbfe3830160e6ba02/dags/iop/parser.py#L310-L316

Reply:
I had a discussion about it with Harris. The initial idea was to have all subfields parsed separately and then joined into one field after all processing steps are done (generic parsing, enchantment, and enrichment. In this PR we have the publisher-specific parser). 
However, I switched to the idea, that maybe it's better to join these fields in the final field already here because, in the [enhancer](https://github.com/cern-sis/workflows/blob/76307b8c4085fd7a41634efb8ec105423bad2542/dags/common/enhancer.py#L15-L23), we are creating unnecessary dict keys and then deleting them if the field value is empty. 
However, then we are creating this big field in every publisher, rather than just to have in one place. So, in the end, we decided to keep subfields separately parsed.

Reply:
ok, sure, but you reference the method by `self` and it's in `Enhancer`

Reply:
the methods referenced by self always are in the parser, because are parser-specific methods. 
I wanted to remove **copyright field** construction from Enhancer and just keep it as part of the publisher-specific parser method. But we moved from this idea. Now we are keeping subfields of copyright parsing: **copyright_year** and **copyright_statement** parsed separately and just the joining then in Enhancer.

The Enhancer change was never done and it had to be on separate PR because it would have affected all publishers. But it won't be done at all, because we moved from the idea of having fields joined in the publisher-specific parser. 

Is it more clear?